events/handler.go

```diff
@@ -0,0 +1,177 @@
+	EventType     EventCategory
+	EventData     EventDataInterface
+}
+
+// Request holds a set of parameters by which Event objects may be screened.
+// The caller may want events that occurred within a specific timeframe
+// or of a certain type, which may be specified in the *Request object
+// they pass to an EventManager function
+type Request struct {
+	StartTime   time.Time
+	EndTime     time.Time
+	EventType   map[EventCategory]bool
+	LastNEvents int
+}
+
+// EventCategory is an enumerated type which lists the categories under which
+// events may fall. The Event field EventType is populated by this enum.
+type EventCategory int
+
+const (
+	OomInstanceCat EventCategory = iota
+	ContainerCreation
+	ContainerDeletion
+)
+
+// a general interface which populates the Event field EventData. The actual
+// object, such as an OomInstance, is set as an Event's EventData
+type EventDataInterface interface {
+}
+
+// returns a pointer to an initialized Events object
+func NewEventManager() *events {
+	return &events{
+		eventlist: make(EventSlice, 0),
+	}
+}
+
+// returns a pointer to an initialized Request object
+func NewRequest() *Request {
+	return &Request{
+		EventType: map[EventCategory]bool{},
+	}
+}
+
+// function necessary to implement the sort interface on the Events struct
+func (e EventSlice) Len() int {
+	return len(e)
+}
+
+// function necessary to implement the sort interface on the Events struct
+func (e EventSlice) Swap(i, j int) {
+	e[i], e[j] = e[j], e[i]
+}
+
+// function necessary to implement the sort interface on the Events struct
+func (e EventSlice) Less(i, j int) bool {
+	return e[i].TimeOfEvent.Before(e[j].TimeOfEvent)
+}
+
+// sorts and returns up to the last N chronological elements
+func getLastNEvents(request *Request, eSlice EventSlice) EventSlice {
+	sort.Sort(eSlice)
+	n := request.LastNEvents
+	if n >= eSlice.Len() || n <= 0 {
+		return eSlice
+	}
+	return eSlice[eSlice.Len()-n:]
+}
+
+// determines if an event occurs within the time set in the request object and is the right type
+func checkIfEventSatisfiesRequest(request *Request, event *Event) bool {
+	startTime := request.StartTime
+	endTime := request.EndTime
+	eventTime := event.TimeOfEvent
+	if !startTime.IsZero() {
+		if startTime.After(eventTime) {
+			return false
+		}
+	}
+	if !endTime.IsZero() {
+		if endTime.Before(eventTime) {
+			return false
+		}
+	}
+	if request.EventType[event.EventType] != true {
+		return false
+	}
+	return true
+}
+
+// method of Events object that screens Event objects found in the eventlist
+// attribute and if they fit the parameters passed by the Request object,
+// adds it to a slice of *Event objects that is returned
+func (self *events) GetEvents(request *Request) (EventSlice, error) {
+	returnEventList := EventSlice{}
+	for _, e := range self.eventlist {
+		if checkIfEventSatisfiesRequest(request, e) {
+			returnEventList = append(returnEventList, e)
+		}
+	}
+	returnEventList = getLastNEvents(request, returnEventList)
```

Comment:
I was thinking if the user specified both, it would return up to N events that fell in the time range.  I could change the field name LastNEvents to MaxEventsReturned or GetNEvents. In case there's a ton of events, it could be helpful to the user to limit the number of events returned.